examples/common.cpp

```diff
@@ -23,17 +24,41 @@ extern "C" __declspec(dllimport) int __stdcall WideCharToMultiByte(unsigned int
 #define CP_UTF8 65001
 #endif
 
-bool gpt_params_parse(int argc, char ** argv, gpt_params & params) {
-    // determine sensible default number of threads.
-    // std::thread::hardware_concurrency may not be equal to the number of cores, or may return 0.
+int32_t get_num_physical_cores() {
 #ifdef __linux__
     std::ifstream cpuinfo("/proc/cpuinfo");
-    params.n_threads = std::count(std::istream_iterator<std::string>(cpuinfo),
-                                  std::istream_iterator<std::string>(),
-                                  std::string("processor"));
+    std::string line;
+    while (std::getline(cpuinfo, line)) {
+        if (line.find("cpu cores") != std::string::npos) {
+            line.erase(0, line.find(": ") + 2);
+            try {
+                return (int32_t) std::stoul(line);
+            } catch (std::invalid_argument& e) {} // Ignore if we could not parse
+        }
+    }
+#elif defined(__APPLE__) && defined(__MACH__)
+    int num_physical_cores;
```

Comment:
Just use `int32` here and you don't need to cast it two times below. 
```suggestion
    int32 num_physical_cores;
```